firewood/src/storage/persisted/mod.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+// TODO: remove this once we have code that uses it
+#![allow(dead_code)]
+
+use bytes::Bytes;
+use std::io::{Error, SeekFrom};
+use std::path::PathBuf;
+use tokio::fs::{File, OpenOptions};
+use tokio::io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt};
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub struct StoreWrite {
+    offset: u64,
+    data: Bytes,
+}
+
+pub struct PersistedStore {
+    path: PathBuf,
+    fd: Mutex<File>,
+}
+
+impl PersistedStore {
+    pub async fn new(path: PathBuf) -> Result<Self, Error> {
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .truncate(false)
+            .create(true)
+            .mode(0o600)
+            .open(&path)
+            .await?;
+        Ok(Self {
+            path,
+            fd: Mutex::new(file),
+        })
+    }
+
+    // TODO: impl ReadLinearStore when it is async
+    pub async fn stream_from(&self, addr: u64, length: usize) -> Result<Bytes, Error> {
```

Comment:
The point of `stream_from` is to return a readable stream (something that implements the `Read` trait or perhaps in your case `AsyncRead`). You can't pass in a length, otherwise this is just a read method.

Reply:
This is intended to replace `FileBacked` in the read path. However as the `LinearStore` signature is sync, will update its signature to match when `LinearStore` switch to async. Which definitely can come later.

Reply:
If that's the case, you'll need to implement the other methods. We don't want a length here. The length is determined by how many bytes they read from the returned trait.